client/apps/edit/components/content/sections/text/index.coffee

```diff
@@ -49,73 +48,119 @@ module.exports = React.createClass
     @refs.editor.focus()
 
-  convertFromHTML: (html) ->
-    blocksFromHTML = convertFromHTML({
-      htmlToStyle: (nodeName, node, currentStyle) ->
-        if nodeName is 'span' and node.style.textDecoration is 'line-through'
-          return currentStyle.add 'STRIKETHROUGH'
-        else
-          return currentStyle
-      htmlToEntity: (nodeName, node) ->
-        if nodeName is 'a'
-          data = {url: node.href, name: node.name, className: node.classList.toString()}
-          return Entity.create(
-              'LINK',
-              'MUTABLE',
-              data
-          )
-        if nodeName is 'p' and node.innerHTML is '<br>'
-          node.innerHTML = '' # remove <br>, it renders extra breaks in editor
-      })(html)
-    return blocksFromHTML
-
-  convertToHtml: (editorState) ->
-    html = convertToHTML({
-      entityToHTML: (entity, originalText) ->
-        if entity.type is 'LINK'
-          sanitizeName = originalText.split(' ')[0].replace(/[.,\/#!$%\^&\*;:{}=\_`’'~()]/g,"")
-          name = if entity.data.name then ' name="' + sanitizeName + '"' else ''
-          if entity.data.className?.includes('is-follow-link')
-            artist = entity.data.url.split('/artist/')[1]
-            return '<a href="' + entity.data.url + '" class="' + entity.data.className + '"' + name + '>' + originalText + '</a><a data-id="'+ artist + '" class="entity-follow artist-follow"></a>'
-          else
-            return a { href: entity.data.url}
-        return originalText
-      blockToHTML: (block) ->
-        if block.type is 'header-three'
-          return h3 {}, block.text
-      styleToHTML: (style) ->
-        if style is 'STRIKETHROUGH'
-          return span { style: {textDecoration: 'line-through'}}
-    })(editorState.getCurrentContent())
-    # put the line breaks back for correct client rendering
-    html = html
-      .replace /<p><\/\p>/g, '<p><br></p>'
-      .replace /<p> <\/\p>/g, '<p><br></p>'
-      .replace(/  /g, ' &nbsp;')
-    html = if html is '<p><br></p>' then '' else html
-    return html
+  setSelectionToStart: (editorState) ->
+    firstKey = editorState.getCurrentContent().getFirstBlock().getKey()
+    newSelection = new SelectionState {
+      anchorKey: firstKey
+      anchorOffset: 0
+      focusKey: firstKey
+      focusOffset: 0
+    }
+    return EditorState.forceSelection editorState, newSelection
+
+  handleReturn: (e) ->
+    selection = getSelectionDetails(@state.editorState)
+    if selection.isFirstBlock or selection.anchorOffset
+      return 'not-handled'
+    else
+      e.preventDefault()
+      @splitSection(selection.anchorKey)
+      return 'handled'
+
+  handleTab: (e) ->
+    e.preventDefault()
+    index = @props.index + 1
+    if e.shiftKey
+      index = @props.index - 1
+    @props.onSetEditing index
+
+  handleBackspace: (e) ->
+    selection = getSelectionDetails(@state.editorState)
+    if selection.isFirstBlock and selection.anchorOffset is 0 and
```

Comment:
I think in places like this, it's worth adding a comment since these conditions don't explain super clearly what's happening. There are a couple other places I'm seeing that are pretty confusing to understand -- mind giving a quick look-through and find opportunities for more comments?